test/kb_Metrics_server_test.py

```diff
@@ -118,6 +187,549 @@ def getImpl(self):
+            self.assertTrue(any(d['k'] == 'narrative_nice_name'
+                                for d in wn['meta']))
+
+        self.assertEqual(len(ws_narrs), 24)
+        self.assertIn('username', ws_narrs[0])
+        self.assertIn('workspace_id', ws_narrs[0])
+        self.assertIn('name', ws_narrs[0])
+        self.assertIn('deleted', ws_narrs[0])
+        self.assertIn('desc', ws_narrs[0])
+        self.assertIn('numObj', ws_narrs[0])
+        self.assertIn('last_saved_at', ws_narrs[0])
+
+        self.assertEqual(ws_narrs[23]['username'], 'psdehal')
+        self.assertEqual(ws_narrs[23]['workspace_id'], 27834)
+        self.assertEqual(ws_narrs[23]['name'],
+                         'psdehal:narrative_1513709108341')
+        self.assertEqual(ws_narrs[23]['meta'][2]['k'], 'narrative_nice_name')
+        self.assertEqual(ws_narrs[23]['meta'][2]['v'], 'Staging Test')
+        self.assertFalse(ws_narrs[23]['deleted'])
+        self.assertEqual(ws_narrs[23]['desc'], '')
+        self.assertEqual(ws_narrs[23]['numObj'], 4)
+        self.assertEqual(ws_narrs[23]['last_saved_at'],
+                         datetime.datetime(2018, 1, 24, 19, 35, 30, 1000))
+
+    # Uncomment to skip this test
+    # @unittest.skip("skipped test_MetricsMongoDBs_list_ujs_results")
+    def test_MetricsMongoDBs_list_ujs_results(self):
+        minTime = 1500000932952
+        maxTime = 1500046845591
+        user_list1 = ['tgu2', 'umaganapathyswork', 'arfath']
+        user_list2 = ['umaganapathyswork', 'arfath']
+
+        epoch = datetime.datetime.utcfromtimestamp(0)
+        # testing list_ujs_results return data, with userIds
+        ujs = self.dbi.list_ujs_results(user_list1, minTime, maxTime)
+        self.assertEqual(len(ujs), 14)
+        ujs = self.dbi.list_ujs_results(user_list2, minTime, maxTime)
+        self.assertEqual(len(ujs), 3)
+        for uj in ujs:
+            self.assertIn(uj.get('user'), user_list2)
+            uj_creation_time = int((uj.get('created') -
+                                    epoch).total_seconds() * 1000.0)
+            self.assertTrue(minTime <= uj_creation_time <= maxTime)
+
+        # testing list_ujs_results return data, without userIds
+        ujs = self.dbi.list_ujs_results([], minTime, maxTime)
+        self.assertEqual(len(ujs), 14)
+
+        # testing list_ujs_results return data, different userIds and times
+        ujs = self.dbi.list_ujs_results(['wjriehl'],
+                                        1500052541065, 1500074641912)
+        self.assertEqual(len(ujs), 8)
+        ujs = self.dbi.list_ujs_results([], 1500052541065, 1500074641912)
+        self.assertEqual(len(ujs), 14)
+
+    # Uncomment to skip this test
+    # @unittest.skip("skipped test_MetricsMongoDBController_constructor")
+    def test_MetricsMongoDBController_constructor(self):
+        cfg_arr = self.cfg
+        # testing if all the required parameters are given
+        with self.assertRaises(ValueError):
```

Comment:
Please see how to properly test raise error here:
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises

Reply:
improved.